examples/split_plot_rcbd.R

```diff
@@ -0,0 +1,159 @@
+                 SeedLotA:TrtmtB, data = my.data)
+#-----------------------------------------------------------------------------#
+
+# Plot the four standard fit plots: residuals vs predicted, sqrt of residuals vs
+# fitted, Normal Q-Q plot of the residuals, residuals vs leverages.
+dev.new()
+#-----------------------------------------------------------------------------#
+par(mfrow = c(2, 2), oma = c(0, 0, 2, 0))
+plot(model.tmp)
+#-----------------------------------------------------------------------------#
+
+# Make sure the residuals are normal (this can also be seen in the Q-Q plot).
+cat('Shapiro-Wilk Normality Test\n')
+sep(50)
+#-----------------------------------------------------------------------------#
+shapiro.test(residuals(model.tmp))
+#-----------------------------------------------------------------------------#
+sep(50)
+
+# Add Levene's Test.
+leveneTest(Yield ~ SeedLotA, data=my.data)
+leveneTest(Yield ~ TrtmtB, data=my.data)
+
+#1-df Tukey:
+my.data$sq_preds <- predict(model.tmp)^2
+one.df.model <- lm(Yield ~ SeedLotA + Block + SeedLotA:Block + TrtmtB + SeedLotA:TrtmtB + sq_preds, my.data)
+anova(one.df.model)
+
+# User selected alpha level.
+#-----------------------------------------------------------------------------#
+alpha <- 0.05
+#-----------------------------------------------------------------------------#
+
+# Print the ANOVA table of the fit. The user should will have to note the
+# significant factors. In this case the single virus factor is significant.
+# NOTE : `anova(model)` fails here because the `model` variable contains a list
+# of anova results due to the `Error()` term.
+cat('ANOVA Table\n')
+sep(50)
+#-----------------------------------------------------------------------------#
+summary(model)
+#-----------------------------------------------------------------------------#
+sep(50)
+#ANOVA table shows significant interaction ==> must do LSD for simple effects (all 12 combinations of SeedLotA and TrtmtB)
+# Show the confidence intervals of the model.
+cat('Confidence Intervals\n')
+sep(50)
+#-----------------------------------------------------------------------------#
+confint(model.tmp, level=1.0 - alpha)
+#-----------------------------------------------------------------------------#
+sep(50)
+
+# Plot the mean yield with respect to each factor.
+dev.new()
+#-----------------------------------------------------------------------------#
+plot(allEffects(model.tmp))
+#-----------------------------------------------------------------------------#
+#plot interaction plots
+library(HH)
+intxplot(Yield ~ SeedLotA, groups = TrtmtB, data=my.data, se=TRUE, ylim=range(my.data$Yield),
+  offset.scale=500)
+intxplot(Yield ~ TrtmtB, groups = SeedLotA, data=my.data, se=TRUE, ylim=range(my.data$Yield),
+  offset.scale=500)
+# TODO : Do we simply check for the SeedLotA in this example? Can we use the
+# LSD.test() on the model without the error term for the correct results?
```

Comment:
Since the interaction SeedLotA x TrtmtB is significant, we CANNOT run the LSD on the two main effects (1) LSD.test(my.data,'SeedLotA') and (2) LSD.test(my.data,'TrtmtB'). Instead, we have to run it on the simple effects (every combination of SeedLotA and TrtmtB).